opal/mca/accelerator/null/accelerator_null_component.c

```diff
@@ -0,0 +1,237 @@
+
+/* Accelerator API's */
+static int accelerator_null_check_addr(const void *addr, int *dev_id, uint64_t *flags);
+static int accelerator_null_memcpy(int dev_id, void *dest, const void *src,
+                            size_t size, opal_accelerator_transfer_type_t type);
+static int accelerator_null_vector_memcpy(int dev_id, void *dest, size_t dpitch,
+                                   const void *src, size_t spitch,
+                                   size_t width, size_t height,
+                                   opal_accelerator_transfer_type_t type);
+static int accelerator_null_memmove(int dev_id, void *dest, const void *src, size_t size,
+                             opal_accelerator_transfer_type_t type);
+static int accelerator_null_malloc(void **ptr, size_t size);
+static int accelerator_null_free(void *ptr);
+static int accelerator_null_get_address_range(const void *ptr, void **base,
+                                              size_t *size);
+
+static bool accelerator_null_is_ipc_enabled(void);
+static size_t accelerator_null_get_handle_size(void);
+static int accelerator_null_get_handle(void *dev_ptr, void **handle);
+static int accelerator_null_open_handle(void *handle, int dev_id,  void **dev_ptr);
+static int accelerator_null_close_handle(void *dev_ptr);
+
+static int accelerator_null_host_register(void *ptr, size_t size);
+static int accelerator_null_host_unregister(void *ptr);
+
+static int accelerator_null_get_device(int *dev_id);
+static int accelerator_null_device_can_access_peer( int *access, int dev1, int dev2);
+
+/*
+ * Instantiate the public struct with all of our public information
+ * and pointers to our public functions in it
+ */
+
+opal_accelerator_null_component_t mca_accelerator_null_component = {{
+
+    /* First, the mca_component_t struct containing meta information
+       about the component itself */
+
+    .base_version =
+        {
+            /* Indicate that we are a accelerator v1.1.0 component (which also
+               implies a specific MCA version) */
+
+            OPAL_ACCELERATOR_BASE_VERSION_1_0_0,
+
+            /* Component name and version */
+
+            .mca_component_name = "null",
+            MCA_BASE_MAKE_VERSION(component, OPAL_MAJOR_VERSION, OPAL_MINOR_VERSION,
+                                  OPAL_RELEASE_VERSION),
+
+            /* Component open and close functions */
+
+            .mca_open_component = accelerator_null_open,
+            .mca_close_component = accelerator_null_close,
+            .mca_register_component_params = accelerator_null_component_register,
+
+        },
+    /* Next the MCA v1.0.0 component meta data */
+    .base_data =
+        {/* The component is checkpoint ready */
+         MCA_BASE_METADATA_PARAM_CHECKPOINT},
+    .accelerator_init = accelerator_null_init,
+    .accelerator_finalize = accelerator_null_finalize,
+}};
+
+opal_accelerator_base_module_t opal_accelerator_null_module =
```

Comment:
Can you update with my asynchronous API's

Reply:
done